src/resources/helpers/generate-method.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const { WorkastInvalidParameterError } = require('../../errors');
+
+/**
+ * Method spec.
+ *
+ * @typedef {Object} MethodSpec
+ * @property {string} method - The HTTP method. One of 'GET', 'HEAD', 'POST', 'PUT', 'DELETE' or 'PATCH'.
+ * @property {string} path - The generic URL path.
+ * */
+
+/**
+ * @summary Generates an API method from the given spec.
+ *
+ * @param {Workast} workast - The Workast SDK instance.
+ * @param {MethodSpec} spec - The method spec.
+ *
+ * @returns {Function} An async function to call the API endpoint.
+ * */
+function generateMethod(workast, spec) {
+  // eslint-disable-next-line func-names
+  return async function (...args) {
+    const urlParamsCount = (spec.path.match(/{\w+}/g) || []).length;
```

Comment:
If `path` is not provided we will have an Unhandled Error that i think we are trying to avoid with the defensive code of `|| [ ]`. Also we are destructuring the spec to access path again in line `31` and calling path.replace without checking if is undefined.
We should save the path in a variable with a default empty string before line 24 in the form of `let path = spec.path || ''`

Reply:
Actually, in the jsdoc it is specified that both `spec.method` and `spec.path` are required. The `|| [ ]` check is there for cases where the path does not include path params (for example `/task/search`).

I think extra validations are not required, since this method is not publicly used.

Reply:
Makes sense, no changes required here.